src/ripple/app/tx/impl/Payment.cpp

```diff
@@ -30,6 +30,19 @@ namespace ripple {
 
 // See https://ripple.com/wiki/Transaction_Format#Payment_.280.29
 
+XRPAmount
+Payment::calculateMaxSpend(STTx const& tx)
+{
+    auto const sendMax = tx[~sfSendMax];
+    auto const saDstAmount(tx.getFieldAmount(sfAmount));
+
+    /* If there's no sfSendMax in XRP, and the sfAmount isn't
+    in XRP, then the transaction can not send XRP. */
+    return sendMax ?
+        (sendMax->native() ? sendMax->xrp() : beast::zero) :
+        (saDstAmount.native() ? saDstAmount.xrp() : beast::zero);
+}
+
```

Comment:
saDstAmount may not be used. the triple ternary can be difficult to read.

Suggest:
    auto const& sendMax = tx[~sfSendMax];
    if(sendMax)
        return sendMax->native() ? sendMax->xrp() : beast::zero;

    /* If there's no sfSendMax in XRP, and the sfAmount isn't
    in XRP, then the transaction can not send XRP. */
    auto const& saDstAmount(tx.getFieldAmount(sfAmount));
    return saDstAmount.native() ? saDstAmount.xrp() : beast::zero;